control-plane/connect-inject/endpoints_controller.go

```diff
@@ -433,9 +425,7 @@ func (r *EndpointsController) createServiceRegistrations(pod corev1.Pod, service
 		Address:   pod.Status.PodIP,
 		Meta:      meta,
 		Namespace: r.consulNamespace(pod.Namespace),
-	}
-	if len(tags) > 0 {
```

Comment:
I didn't see why we were doing this `if` check rather than just assigning to the struct field.